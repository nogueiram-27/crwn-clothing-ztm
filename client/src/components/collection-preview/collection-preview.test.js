import React from 'react'
import { shallow } from 'enzyme'
import { CollectionPreview } from './collection-preview.component'
import CollectionItem from '../collection-item/collection-item.component'

describe ('CollectionPreview component', () => {
    const mockTitle = 'hats'
    const mockItems = [ { id: 1}, { id: 2}, { id: 3}, { id: 4}, { id: 5}, { id: 6} ]
    const mockRouteName = 'hats'
    const mockMatch = {
        url: '/shop'
    }
    let mockHistory
    let wrapper

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        }
        const mockProps = {
            title: mockTitle,
            items: mockItems,
            match: mockMatch,
            history: mockHistory,
            routeName: mockRouteName
        }
        wrapper = shallow(<CollectionPreview {...mockProps}/>)
    })

    it('should render CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('trigger actions when click in the title', () => {
        wrapper.find('TitleContainer').simulate('click')
        expect(mockHistory.push).toHaveBeenCalledWith('/shop/hats')
    })

    it('should render CollectionItem only 4 times max', () => {
        expect(wrapper.find(CollectionItem).length).toBe(4)
    })
})