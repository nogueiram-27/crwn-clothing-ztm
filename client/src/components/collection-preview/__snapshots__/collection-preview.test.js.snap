// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render CartItem component 1`] = `ShallowWrapper {}`;

exports[`CollectionPreview component should render CollectionPreview component 1`] = `ShallowWrapper {}`;
